tikv-client/src/main/java/com/pingcap/tikv/Snapshot.java

```diff
@@ -66,14 +68,29 @@ public ByteString get(ByteString key) {
         .get(key, timestamp.getVersion());
   }
 
+  public Iterator<TiColumnarChunk> tableReadChunk(TiDAGRequest dagRequest, long physicalId) {
+    return tableReadChunk(
+        dagRequest,
+        RangeSplitter.newSplitter(session.getRegionManager())
+            .splitRangeByRegion(dagRequest.getRangesByPhysicalId(physicalId)));
+  }
+
+  public Iterator<TiColumnarChunk> tableReadChunk(TiDAGRequest dagRequest, List<RegionTask> tasks) {
+    if (dagRequest.isDoubleRead()) {
+      throw new UnsupportedOperationException(
+          "double read case should first read handle which is rowwise");
+    } else {
+      return getColumnarBatchIterator(dagRequest, tasks, getSession());
+    }
+  }
   /**
    * Issue a table read request
    *
    * @param dagRequest DAG request for coprocessor
    * @return a Iterator that contains all result from this select request.
    */
-  public Iterator<Row> tableRead(TiDAGRequest dagRequest, long physicalId) {
-    return tableRead(
+  public Iterator<Row> tableReadRow(TiDAGRequest dagRequest, long physicalId) {
```

Comment:
Is this rename back-compatible?

Reply:
Checked with @ilovesoup, we only need make sure tispark working as expected.